src/core/positioning/bluetoothreceiver.cpp

```diff
@@ -161,45 +166,18 @@ void BluetoothReceiver::doConnectDevice()
   repairDevice( QBluetoothAddress( mAddress ) );
 }
 
-void BluetoothReceiver::setSocketState( const QBluetoothSocket::SocketState socketState )
+QString BluetoothReceiver::socketStateString()
 {
-  if ( mSocketState == static_cast<QAbstractSocket::SocketState>( socketState ) )
+  const QAbstractSocket::SocketState currentState = socketState();
+  QString socketStateString = AbstractGnssReceiver::socketStateString();
+  if ( currentState == QAbstractSocket::UnconnectedState )
   {
-    return;
-  }
-
-  switch ( socketState )
-  {
-    case QBluetoothSocket::SocketState::ServiceLookupState:
-    case QBluetoothSocket::SocketState::ConnectingState:
+    if ( !mDisconnecting && mSocket->error() != QBluetoothSocket::SocketError::NoSocketError )
```

Comment:
Why do we have a mDisconnecting variable? Id say we should replace it with `mSocket->state() != QBluetoothSocket::SocketState::ClosingState`